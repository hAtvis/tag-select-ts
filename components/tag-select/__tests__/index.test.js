import React from 'react'
import { render, mount } from 'enzyme'
import TagSelect from '..'
import sinon from 'sinon'

const options = [{
  id: 1,
  text: 'John'
}, {
  id: 2,
  text: 'Smith'
}]
describe('tag-select', () => {
  it('render correct', () => {
    const wrapper = render(
      <TagSelect options={options} />
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('default prop multiple is false', () => {
    const wrapper = mount(<TagSelect options={options} />)
    expect(wrapper.props().multiple).toBe(false)
  })

  it('single select click onChange correct', () => {
    const onChange = sinon.spy()
    const wrapper = mount(<TagSelect options={options} onChange={onChange} />) 

    wrapper.find('.ant-tag').first().simulate('click')
    expect(onChange.calledWith(1)).toBe(true)
  })

  it('multiple select click onChange correct', () => {
    const onChange = sinon.spy()
    const wrapper = mount(<TagSelect options={options} onChange={onChange} multiple={true} />)

    wrapper.find('.ant-tag').first().simulate('click')
    expect(onChange.calledWith([1])).toBe(true)
  })

  it('single check', () => {
    const wrapper = render(<TagSelect options={options} defaultValue={1} />)
    expect(wrapper).toMatchSnapshot()
  })
})