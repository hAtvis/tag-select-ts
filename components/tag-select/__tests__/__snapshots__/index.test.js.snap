// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tag-select render correct 1`] = `
Array [
  <div
    class="ant-tag ant-tag-checkable"
  >
    John
  </div>,
  <div
    class="ant-tag ant-tag-checkable"
  >
    Smith
  </div>,
]
`;

exports[`tag-select single check 1`] = `
Array [
  <div
    class="ant-tag ant-tag-checkable ant-tag-checkable-checked"
  >
    John
  </div>,
  <div
    class="ant-tag ant-tag-checkable"
  >
    Smith
  </div>,
]
`;
